### PR TITLE
Add chain connector plugin

### LINKED ISSUE #736: Add chain connector plugin
### Description

Add chain connector plugin so that relayer node to automatically create and submit Cross Chain Transaction by aggregating off-chain information of a chain.

### Motivation

- Aggregate information to create cross chain transaction
- To meet protocol liveliness requirements

### PR DESCRIPTION
**Pull request by ManuGowda**
### What was the problem?

This PR resolves #736 , #738

### How was it solved?

- Add chain connector plugin
- Update genesis block
- Update legacy store add index

### How was it tested?

- Run `./bin/run start -n devnet --api-ipc -l debug --api-ws`

### CODE REVIEW

File: src/application/modules/legacy/module.ts [hunk 1]
```diff
@@ -60,7 +60,7 @@ export class LegacyModule extends BaseModule {
 
 	public constructor() {
 		super();
-		this.stores.register(LegacyAccountStore, new LegacyAccountStore(this.name));
+		this.stores.register(LegacyAccountStore, new LegacyAccountStore(this.name, 0));
 		this.events.register(ReclaimLSKEvent, new ReclaimLSKEvent(this.name));
```
Comment (sameersubudhi) [inline src/application/modules/legacy/module.ts:64]: Could we please add an empty line for visual clarity in separation of store and event registration.
```suggestion
		this.stores.register(LegacyAccountStore, new LegacyAccountStore(this.name, 0));

		this.events.register(ReclaimLSKEvent, new ReclaimLSKEvent(this.name));
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Code changes look good. Could you please confirm if the alphanet/devnet genesis blocks were recreated post the change made in `src/application/modules/legacy/module.ts`? If I run through the following steps locally a different genesis block gets generated for both the networks.

- `nvm use`
- `npm ci`
- `npm run build`
- `./bin/run genesis-block create -n devnet -o ./config/devnet -f ./config/devnet/genesis_assets.json`
- `./bin/run genesis-block create -n alphanet -o ./config/alphanet -f ./config/alphanet/genesis_assets.json`

File: src/application/modules/legacy/module.ts (hunk 1 above)
Comment (sameersubudhi) [reply src/application/modules/legacy/module.ts:64]: Also, seeing the following error when running the node with the chain connector enabled:

Start command:
```
./bin/run start -n devnet --api-ipc -l debug --api-ws --enable-chain-connector-plugin
```

Logged error:
```
2023-02-27T09:16:56.764Z DEBUG Sameers-MacBook-Pro.local plugin_chainConnector 27584 [plugin=chainConnector] Plugin is registered to bus
2023-02-27T09:16:56.767Z ERROR Sameers-MacBook-Pro.local plugin_chainConnector 27584 [err=Lisk validator found 3 error[s]:
Missing property, must have required property 'ccuFee'
Missing property, must have required property 'encryptedPrivateKey'
Missing property, must have required property 'password'] Fail to handle message.
```

Comment (ManuGowda) [reply src/application/modules/legacy/module.ts:64]: I think all of these are optional config params? @shuse2 is this correct?

**Review by sameersubudhi — APPROVED** [review]
LGTM 👍

**Review by nagdahimanshu — APPROVED** [review]
Looks good 👍

**Pull request merged**